src/geometry.F90

```diff
@@ -610,13 +611,33 @@ subroutine cross_surface(p, last_cell)
       ! cells on the positive side
 
       call find_cell(p, found, surf%neighbor_pos)
+
+      ! Save cosine of angle between surface normal and particle direction
+      p % normal_proj = dot_product(p % coord(1) % uvw, &
+                  & surf % normal(p % coord(1) % xyz)) / &
+                  & sqrt(dot_product(surf % normal(p % coord(1) % xyz)&
+                  &, surf % normal(p % coord(1) % xyz)))
+
+      ! Score cell to cell partial currents
+      call score_partial_current(p)
```

Comment:
Good idea. I put it there to be with score_partial_current, but it s less code in less locations to have it in tracking